singularityhub/sregistry
# Update to container push to honor new binary

This pull request will address #273 - specifically, we had a typo in the final tag function where, given that a container was not frozen, we would want to keep the newly uploaded one and clean up the previous one. In addition, the django signal wasn't firing to clean up any image files that were no longer used.

To test this pull request:

 - keep in mind that since the docker-compose has a parameter added to postgres (container df) you should not test this on a production running instance, because there is a chance of the container being re-created and losing the database.
 - if you are testing locally you will need to follow instructions to compile a singularity version that changes a line in remote.go to http from https (details [here](https://singularityhub.github.io/sregistry/docs/client)) and also there are detail for how to define and login to the remote. Don't forget to generate your secrets.py file with a secret_key!

## Things to test
- [ ] creating a new collection and pushing a container
- [ ] push of a container with a new tag
- [ ] push of a container with an existing tag
- [ ] push of the same container with a different tag

I tested these points, and I'd like one more tester for a sanity check. @RonaldEnsing I'm hoping since the issue is important to you that you might be able to - and please there is no rush! Take whatever time that you need. 

Signed-off-by: vsoch <vsochat@stanford.edu>
